### PR TITLE
Model Parser: JavaDeclaration and JavaMethod code generators implemented.

### PR DESCRIPTION
**Pull request by azabbasi**

### CODE REVIEW

**Review by YoDaMa — APPROVED** [review]
lgtm

File: sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaDeclaration.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    public void addAttributes(String text) {
+        this.attributes.add(text);
+    }
+
+    /**
+     * Generate code for the declaration.
+     *
+     * @param codeWriter A {@link CodeWriter} object for generating the declaration code.
+     */
+    public void generateCode(CodeWriter codeWriter) throws IOException {
```
Comment (barustum) [inline sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaDeclaration.java:181]: No method body. Expected?

**Review by barustum — APPROVED** [review]

**Review by olivakar — APPROVED** [review]

File: sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaDeclaration.java (hunk 1 above)
Comment (azabbasi) [reply sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaDeclaration.java:181]: yes, the subclasses of declaration will override it and they know how to generate the code.

Comment (azabbasi) [reply sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaDeclaration.java:181]: look at `JavaMethod` for example.

File: sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaDeclaration.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+     * Gets the value of the name decorated with appropriate keywords.
+     *
+     * @return The value of the name decorated with appropriate keywords.
+     */
+    public String getDecoratedName() {
+        StringBuilder decoratedName = new StringBuilder();
+
+        switch (access) {
+            case IMPLICIT:
+            case PACKAGE_PRIVATE:
```
Comment (drwill-ms) [inline sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaDeclaration.java:105]: comment as to why no action is taken?

**Pull request merged**